app/controllers/guests_controller.rb

```diff
@@ -0,0 +1,45 @@
+class GuestsController < ApplicationController
+  before_action :require_login, only: [:new, :create]
+
+  def show
+
+  end
+
+  def new
+    @guest = Guest.new
+    respond_to do |format|
+      if request.xhr?
+        format.html { render :new, layout: false}
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def create
+    new_guest = Guest.new(get_params)
```

Comment:
Why "get_params" - get is a really overloaded term in these apps. Get suggests to the reader that you are dealing with a getter, or a get request. Calling this get_params makes the reader pause and figure out what's going on. You should if possible follow the principle of least astonishment.